client/packages/system/src/Stock/api/hooks/useInventoryAdjustment.ts

```diff
@@ -0,0 +1,86 @@
+import { useCallback, useState } from 'react';
+import {
+  AdjustmentDirectionInput,
+  useAuthContext,
+  useGql,
+  useMutation,
+  useQueryClient,
+} from '@openmsupply-client/common';
+import {
+  InventoryAdjustmentReasonRowFragment,
+  StockLineRowFragment,
+} from '../../..';
+import { getSdk } from '..';
+
+type DraftInventoryAdjustment = {
+  direction: AdjustmentDirectionInput | null;
+  reason: InventoryAdjustmentReasonRowFragment | null;
+  adjustBy: number;
+  newNumberOfPacks: number;
+};
+
+export function useInventoryAdjustment(stockLine: StockLineRowFragment) {
```

Comment:
Also used this as a bit of a start for new API hook ideas

Reply:
Happy with either way, I guess i would only object doing it now if it slows down this PR too much